firebase_test.go

```diff
@@ -350,6 +568,34 @@ func (t *testTokenSource) Token() (*oauth2.Token, error) {
 	}, nil
 }
 
+func overwriteEnv(varName, newVal string, t *testing.T) (string, bool) {
```

Comment:
To make this simpler, could we treat the empty string the same as unset? I think the amend function already does so. Then we wouldn't need the bool value.

Reply:
Yeah, I guess if the user had explicitly set export FIREBASE_CONFIG=""
They'd like an error message.

Reply:
This has the downside that if for any reason the user had 
export FIREBASE_CONFIG=""
after the tests FIREBASE_CONFIG will be unset .
I am ok with this